Gordon360/Services/HousingService.cs

```diff
@@ -19,6 +19,22 @@ public HousingService(IUnitOfWork unitOfWork)
             _unitOfWork = unitOfWork;
         }
 
+        /// <summary>
+        /// Calls a stored procedure that returns the count of rows in the staff whitelist which have the given user id
+        /// (the count should be 0 or 1) and converts that count into boolean
+        /// </summary>
+        /// <param name="userId"> The id of the person using the page </param>
+        /// <returns> Whether or not the user is on the staff whitelist </returns>
+        public bool CheckHousingStaff(string userId)
+        {
+            IEnumerable<ApartmentAppIsStaffViewModel> idResult = null;
+
+            var idParam = new SqlParameter("@USER_ID", userId);
+
+            idResult = RawSqlQuery<ApartmentAppIsStaffViewModel>.query("GET_AA_USER_IS_STAFF @USER_ID", idParam);
+            return idResult.ElementAt(0).ResultCount != 0;
```

Comment:
It is recommended to use something like `throw new HttpException(404, "Not found");` if userId is not found on the whitelist, in compliance with RESTful design

Reply:
I have left a suggestion comment below with a possible way to implement this change. The `HttpNotFound` should go in the HousingController, not the HousingService, so I think this particular file could remain unchanged.

As an unintended benefit, this change would also make the frontend-backend interaction much easier to troubleshoot in the future, and it makes the UI handling of the response much easier